hr_holidays_leave_request_wizard/wizards/hr_leave_wizard.py

```diff
@@ -0,0 +1,553 @@
+# Copyright 2018-2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import babel.dates
+from datetime import datetime, time
+from dateutil.relativedelta import relativedelta
+from pytz import timezone, utc
+
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+from odoo.tools import float_utils
+from odoo.addons.resource.models.resource_mixin import ROUNDING_FACTOR
+
+
+class HrLeaveWizard(models.TransientModel):
+    _name = 'hr.leave.wizard'
+    _description = 'HR Leave Wizard'
+
+    employee_id = fields.Many2one(
+        string='Employee',
+        comodel_name='hr.employee',
+        required=True,
+        default=lambda self: self._default_employee_id(),
+    )
+    name = fields.Char(
+        string='Description',
+    )
+    leave_type_id = fields.Many2one(
+        string='Leave Type',
+        comodel_name='hr.leave.type',
+        required=True,
+        domain=[('valid', '=', True)],
+    )
+    leave_type_request_unit = fields.Selection(
+        related='leave_type_id.request_unit',
+    )
+    date_from = fields.Date(
+        string='Start Date',
+        required=True,
+        default=fields.Date.context_today,
+    )
+    date_to = fields.Date(
+        string='End Date',
+        required=True,
+        default=fields.Date.context_today,
+    )
+    day_ids = fields.One2many(
+        string='Days',
+        comodel_name='hr.leave.wizard.day',
+        inverse_name='wizard_id',
+    )
+    interval_ids = fields.One2many(
+        string='Intervals',
+        comodel_name='hr.leave.wizard.day.interval',
+        inverse_name='wizard_id',
+    )
+
+    def _default_employee_id(self):
+        return self.env.context.get('default_employee_id') or \
+            self.env['hr.employee'].search([
```

Comment:
`self.env.user.employee_ids[:1]` to skip search

Reply:
I'm not sure why it's done like that, but such code appears in a lot of places in Odoo itself